feat: add Injury dashboard

SammanBaidya opened the pull request:
# Description
Please include the exact bug/functionality description and a summary of the changes/ related issues. Please also include any other relevant motivation and context:
Fixes # (bug list priority high/medium/low x.y.z)
Or Implements # (WBS) 

## Related PRS (if any):
This frontend PR is related to the #XXX backend PR.
To test this backend PR you need to checkout the #XXX frontend PR.
…

## Main changes explained:
- Delete file A for removing unused components …
- Update file B for including new pattern …
- Create file C for introducing new components …
…

## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. Clear site data/cache
4. log as admin user
5. go to dashboard→ Tasks→ task→…
6. verify function “A” (feel free to include screenshot here)
7. verify this new feature works in [dark mode](https://docs.google.com/document/d/11OXJfBBedK6vV-XvqWR8A9lOH0BsfnaHx01h1NZZXfI)

## Screenshots or videos of changes:

## Note:
Include the information the reviewers need to know.

rahul07bagul reviewed (changes requested):
Needs some changes on UI, below are the details.

- Dark mode is not working
![Screenshot 2025-06-06 085111](https://github.com/user-attachments/assets/f82ab9e2-2131-4f72-8968-530c9f6e7aa7)

- Show labels for X and Y- axis, Injury Count vs Projects
![Screenshot 2025-06-06 085216](https://github.com/user-attachments/assets/02ed9a0c-db08-4385-815b-305eb3131c60)

- Width of filters.
 If we can fix the width of filters, so in case if new filters values are added then current filter input width might increase, instead of that we can add fix width.
 
![Screenshot 2025-06-06 085613](https://github.com/user-attachments/assets/00bec688-4369-49c0-9daa-737efa54c23e)

Like this: [See this example](https://www.google.com/url?sa=i&url=https%3A%2F%2Fdesign.procore.com%2Fmulti-select&psig=AOvVaw2l1ke00TpDea2y2mJ2QcvW&ust=1749300949757000&source=images&cd=vfe&opi=89978449&ved=0CBQQjRxqFwoTCKDjwpvt3I0DFQAAAAAdAAAAABAE)

- Also color of input filters looks like they are disabled, if possible we can make them as black and then placeholder/label. 

![Screenshot 2025-06-06 090345](https://github.com/user-attachments/assets/997275db-5ed0-4e09-a02c-6762a32a8282)

TaariqMansurie reviewed (changes requested):
Yes I completely agree with what Rahul said above, I tested the changes myself but I would just request the same changes on the UI part. Those include : 
1) dark mode is not working 
![injury severity by projects dark mode not working](https://github.com/user-attachments/assets/90b46802-268f-4e7d-9af9-c37f1e534c8c)

2) Show labels for X and Y- axis, Injury Count vs Projects
![injury severity by projects](https://github.com/user-attachments/assets/272e1f0a-9d83-49bc-a381-ae36404f9805)

3) Hint text of input filters for projects and others is not clearly visible
![injury severity by projects dark mode not working placeholder text not visible](https://github.com/user-attachments/assets/41b09bb3-a3fd-433d-b6cb-78a1d4b40d67)

1709abhishek commented:
My suggestions for the following PR :-
1. To keep the filter width as limited size we can use something like : 
`min-width: 150px; max-width: 300px; overflow: hidden; text-overflow: ellipsis`
Also, we can use something like +1 when selecting more than 2 options in the dropdown.
2. 
<img width="242" alt="Screenshot 2025-06-07 at 11 07 18 AM" src="https://github.com/user-attachments/assets/318e3868-a8c0-4270-ba13-41621f3d5a95" />
As shown in screenshot above, we can use a darker text for the dropdown arrows and calendar font icons. Also, the border can also have a darker shade.

knsspsj19 commented:
Nice work!
But here are some considerations:
1. The X- axis of the graph is static and does not evolve as Project filters are changed. This provides redundant info and clutters up the space. Consider making it dynamic.
![image](https://github.com/user-attachments/assets/e067ce1f-f02e-449a-b04c-4cec538f66fc)

2. Although 4 department filters have been selected on top right. The bar graph does not make explicit categorical distinction between the departments. Consider adding "Hue" to the graph.
![image](https://github.com/user-attachments/assets/cf1e6063-6d5e-4f12-95d0-88156e957390)

pranavgovi reviewed (changes requested):
The layout looks great and the filters + legend are clear.
However, the chart doesn’t render even after selecting multiple filters — could be a data issue or maybe something with the rendering logic.
![WhatsApp Image 2025-07-05 at 11 48 23_fa355aa7](https://github.com/user-attachments/assets/124af15f-0237-455f-a038-a2b6d758d774)

Varshakaranam18 commented:
Even after selecting the filters , The chart is not being rendered. I have accused it using owner account and have also pulled appropriate backend pr
<img width="1683" alt="Screenshot 2025-07-05 at 11 45 02 AM" src="https://github.com/user-attachments/assets/b7774f8a-c1df-4291-866f-29a8336757e6" />

SammanBaidya commented:
> Needs some changes on UI, below are the details.
> 
> * Dark mode is not working
>   ![Screenshot 2025-06-06 085111](https://private-user-images.githubusercontent.com/55353026/452309393-f82ab9e2-2131-4f72-8968-530c9f6e7aa7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTI2NzM1MTIsIm5iZiI6MTc1MjY3MzIxMiwicGF0aCI6Ii81NTM1MzAyNi80NTIzMDkzOTMtZjgyYWI5ZTItMjEzMS00ZjcyLTg5NjgtNTMwYzlmNmU3YWE3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA3MTYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNzE2VDEzNDAxMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTdlMWU3Mzg1NWNiMjBiZWM2YTcxYjBiNDU0YWI1YzVjOGQ2ZWRjZWM2NzIxNjA0M2RkYThkY2U3M2ZjMDYzMzUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.y771tuXS-Ai9DwdZpZCahIRbS8suF6Oxdbf2V-_Le_c)
> * Show labels for X and Y- axis, Injury Count vs Projects
>   ![Screenshot 2025-06-06 085216](https://private-user-images.githubusercontent.com/55353026/452309684-02ed9a0c-db08-4385-815b-305eb3131c60.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTI2NzM1MTIsIm5iZiI6MTc1MjY3MzIxMiwicGF0aCI6Ii81NTM1MzAyNi80NTIzMDk2ODQtMDJlZDlhMGMtZGIwOC00Mzg1LTgxNWItMzA1ZWIzMTMxYzYwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA3MTYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNzE2VDEzNDAxMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWYzZmUxOGYxNWZiZDNkZTJmMjNiOGQ1NjFkODBjNWMzMTc3ZGY2ODkyN2QwOWRkNTQ5ZjIzOWQ4NjEyYzRjNmEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.Z-U55Hk2T5uAnLSD7NKH610QBGzbs0aHrT06mgsKS-k)
> * Width of filters.
>   If we can fix the width of filters, so in case if new filters values are added then current filter input width might increase, instead of that we can add fix width.
> 
> ![Screenshot 2025-06-06 085613](https://private-user-images.githubusercontent.com/55353026/452310394-00bec688-4369-49c0-9daa-737efa54c23e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTI2NzM1MTIsIm5iZiI6MTc1MjY3MzIxMiwicGF0aCI6Ii81NTM1MzAyNi80NTIzMTAzOTQtMDBiZWM2ODgtNDM2OS00OWMwLTlkYWEtNzM3ZWZhNTRjMjNlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA3MTYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNzE2VDEzNDAxMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY4Y2ZjZDljMTQ0ZTZiOTIyMmVjMDY3MWUwMTYzNmFkNjFhNmFiZGJiNDI5YjBiZDAwYjRlYzdhMzFiNmMyNGEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.tX8d0WVarGv-P-YNyHvqYmPguUM4HMMfqcQNY2LQ3vI)
> 
> Like this: [See this example](https://www.google.com/url?sa=i&url=https%3A%2F%2Fdesign.procore.com%2Fmulti-select&psig=AOvVaw2l1ke00TpDea2y2mJ2QcvW&ust=1749300949757000&source=images&cd=vfe&opi=89978449&ved=0CBQQjRxqFwoTCKDjwpvt3I0DFQAAAAAdAAAAABAE)
> 
> * Also color of input filters looks like they are disabled, if possible we can make them as black and then placeholder/label.
> 
> ![Screenshot 2025-06-06 090345](https://private-user-images.githubusercontent.com/55353026/452311255-997275db-5ed0-4e09-a02c-6762a32a8282.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTI2NzM1MTIsIm5iZiI6MTc1MjY3MzIxMiwicGF0aCI6Ii81NTM1MzAyNi80NTIzMTEyNTUtOTk3Mjc1ZGItNWVkMC00ZTA5LWEwMmMtNjc2MmEzMmE4MjgyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA3MTYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNzE2VDEzNDAxMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWI4MjdiYjA2MDdiZDI3OGM0ZmUzN2ZlOTI0MmNlNzRmYTIxNTgzZGU0NDNiZTJkZWRlYjFiOWNjY2EwZmIzZDMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.9SwCF48Spiowmj-kq_rPE16hm4FqTudkz7-VfAxk70c)

Can you re-review it!

SammanBaidya commented:
> Yes I completely agree with what Rahul said above, I tested the changes myself but I would just request the same changes on the UI part. Those include :
> 
> 1. dark mode is not working
>    ![injury severity by projects dark mode not working](https://private-user-images.githubusercontent.com/70195858/452547191-90b46802-268f-4e7d-9af9-c37f1e534c8c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTMxNjIxODIsIm5iZiI6MTc1MzE2MTg4MiwicGF0aCI6Ii83MDE5NTg1OC80NTI1NDcxOTEtOTBiNDY4MDItMjY4Zi00ZTdkLTlhZjktYzM3ZjFlNTM0YzhjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA3MjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNzIyVDA1MjQ0MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY2OThmNDIxMWViMGQ0ZjE0NTc5MzhhYjJiNWE5ZGNkZmE3ZDA5ZDBjNTAxMTMyYjcwMjhiZGNlY2JhMmZjN2EmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.6GOG21ee0siO0eaDLbIR2pjfZvB53bJUVLyoz3XE4Ms)
> 2. Show labels for X and Y- axis, Injury Count vs Projects
>    ![injury severity by projects](https://private-user-images.githubusercontent.com/70195858/452547204-272e1f0a-9d83-49bc-a381-ae36404f9805.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTMxNjIxODIsIm5iZiI6MTc1MzE2MTg4MiwicGF0aCI6Ii83MDE5NTg1OC80NTI1NDcyMDQtMjcyZTFmMGEtOWQ4My00OWJjLWEzODEtYWUzNjQwNGY5ODA1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA3MjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNzIyVDA1MjQ0MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTc3ODRmZjg0M2IyN2U1OTQwOGM2Mjg5NGM5YjU0OGUwNWEzNzUxOTg0M2Q5YmE4MjFhZTNjZWZmODVhZDJiMjEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.4QKQJU7uqDcIaVym1uVKXpNncQM1Nb2aEkP3w1NfAqk)
> 3. Hint text of input filters for projects and others is not clearly visible
>    ![injury severity by projects dark mode not working placeholder text not visible](https://private-user-images.githubusercontent.com/70195858/452547176-41b09bb3-a3fd-433d-b6cb-78a1d4b40d67.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTMxNjIxODIsIm5iZiI6MTc1MzE2MTg4MiwicGF0aCI6Ii83MDE5NTg1OC80NTI1NDcxNzYtNDFiMDliYjMtYTNmZC00MzNkLWI2Y2ItNzhhMWQ0YjQwZDY3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA3MjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNzIyVDA1MjQ0MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTk1OTZiZmRkMjUxYzZjMzZlYmM5OTRmYTk2ODZkMGI2YjY4NWYzNmYzYzVjN2RlZDI3NzA4M2IwZmE3NGQ4ZDEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.YCs7Imkfbe97Ayg7sVB6vokaLeb7LkWmaJ7q_Yzahis)

Fixed and updated the code, Please re-review!
